code/__HELPERS/datastructs/armor.dm

```diff
@@ -1,3 +1,234 @@
+	UNPACK_OR(laser, ARMOR_LASER, 0)
+	UNPACK_OR(laser_tier, ARMOR_LASER_TIER, ARMOR_TIER_DEFAULT)
+	UNPACK_OR(laser_soak, ARMOR_LASER_SOAK, 0)
+	UNPACK_OR(energy, ARMOR_ENERGY, 0)
+	UNPACK_OR(bomb, ARMOR_BOMB, 0)
+	UNPACK_OR(bio, ARMOR_BIO, 0)
+	UNPACK_OR(rad, ARMOR_RAD, 0)
+	#undef UNPACK_OR
+
+/datum/armor/proc/to_list()
+	return list(
+		ARMOR_MELEE = melee,
+		ARMOR_MELEE_TIER = melee_tier,
+		ARMOR_MELEE_SOAK = melee_soak,
+		ARMOR_BULLET = bullet,
+		ARMOR_BULLET_TIER = bullet_tier,
+		ARMOR_BULLET_SOAK = bullet_soak,
+		ARMOR_LASER = laser,
+		ARMOR_LASER_TIER = laser_tier,
+		ARMOR_LASER_SOAK = laser_soak,
+		ARMOR_ENERGY = energy,
+		ARMOR_BOMB = bomb,
+		ARMOR_BIO = bio,
+		ARMOR_RAD = rad,
+	)
+
+/datum/armor/proc/to_name()
+	return jointext(list(
+		"[melee]@[melee_tier]-[melee_soak]",
+		"[bullet]@[bullet_tier]-[bullet_soak]",
+		"[laser]@[laser_tier]-[laser_soak]",
+		"[energy]",
+		"[bomb]",
+		"[bio]",
+		"[rad]",
+	), "-")
+
+/datum/armor/proc/raw(flag)
+	switch(flag)
+		if(ARMOR_MELEE)
+			return melee
+		if(ARMOR_MELEE_SOAK)
+			return melee_soak
+		if(ARMOR_MELEE_TIER)
+			return melee_tier
+		if(ARMOR_BULLET)
+			return bullet
+		if(ARMOR_BULLET_SOAK)
+			return bullet_soak
+		if(ARMOR_BULLET_TIER)
+			return bullet_tier
+		if(ARMOR_LASER)
+			return laser
+		if(ARMOR_LASER_SOAK)
+			return laser_soak
+		if(ARMOR_ENERGY)
+			return energy
+		if(ARMOR_BOMB)
+			return bomb
+		if(ARMOR_BIO)
+			return bio
+		if(ARMOR_RAD)
+			return rad
+
+/datum/armor/proc/mitigation(flag, tier = ARMOR_TIER_DEFAULT)
+	switch(flag)
+		if(ARMOR_MELEE)
+			var/tdiff = melee_tier - tier
+			return 1 - ARMOR_TIER_CALC(melee, tdiff)
+		if(ARMOR_BULLET)
+			var/tdiff = bullet_tier - tier
+			return 1 - ARMOR_TIER_CALC(bullet, tdiff)
+		if(ARMOR_LASER)
+			var/tdiff = laser_tier - tier
+			return 1 - ARMOR_TIER_CALC(laser, tdiff)
+		if(ARMOR_ENERGY)
+			return energy
+		if(ARMOR_BOMB)
+			return bomb
+		if(ARMOR_BIO)
+			return bio
+		if(ARMOR_RAD)
+			return rad
+
+/datum/armor/proc/soak(flag)
+	switch(flag)
+		if(ARMOR_MELEE)
+			return melee_soak
+		if(ARMOR_BULLET)
+			return bullet_soak
+		if(ARMOR_LASER)
+			return laser_soak
+		else
+			return 0
+
+/datum/armor/proc/resultant_damage(damage, tier, flag)
+	switch(flag)
+		if(ARMOR_MELEE)
+			var/tdiff = melee_tier - tier
+			return max(0, (tdiff? (damage * (1 - ARMOR_TIER_CALC(melee, tdiff))) : (damage * (1 - melee))) - melee_soak)
+		if(ARMOR_BULLET)
+			var/tdiff = bullet_tier - tier
+			return max(0, (tdiff? (damage * (1 - ARMOR_TIER_CALC(bullet, tdiff))) : (damage * (1 - bullet))) - bullet_soak)
+		if(ARMOR_LASER)
+			var/tdiff = laser_tier - tier
+			return max(0, (tdiff? (damage * (1 - ARMOR_TIER_CALC(laser, tdiff))) : (damage * (1 - laser))) - laser_soak)
+		if(ARMOR_ENERGY)
+			return damage * (1 - energy)
+		if(ARMOR_BOMB)
+			return damage * (1 - bomb)
+		if(ARMOR_BIO)
+			return damage * (1 - bio)
+		if(ARMOR_RAD)
+			return damage * (1 - rad)
+
+/datum/armor/proc/describe_list()
```

Comment:
I assume this is pseudo OOC information

Reply:
i don't really like the way i'm handling this but it's the only sane way i could think of
i liked the old system where it has 10 strings for 10 cases, but

- handling tiers would be weird
- handling soak would be weird because we might have armor with shit mitigation and very good soak later